File: user-service/src/main/java/com/example/user/service/UserService.java
package com.example.user.service;

import com.example.user.entity.User;
import com.example.user.repository.UserRepository;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    public User createUser(CreateUserRequest request) {
        User user = new User(request.getEmail(), request.getName());
        return userRepository.save(user);
    }

    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    public User getUserById(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("User not found with id: " + id));
    }

    public Optional<User> getUserByEmail(String email) {
        return userRepository.findByEmail(email);
    }

    @Getter
    @Setter
    @NoArgsConstructor
    public static class CreateUserRequest {
        private String email;
        private String name;

        public CreateUserRequest(String email, String name) {
            this.email = email;
            this.name = name;
        }
    }
}

File: src/main/kotlin/com/example/monolith/user/UserService.kt
package com.example.monolith.user

import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository
) {
    fun createUser(request: CreateUserRequest): User {
        val user = User(
            email = request.email,
            name = request.name
        )
        return userRepository.save(user)
    }

    fun getAllUsers(): List<User> {
        return userRepository.findAll()
    }

    fun getUserById(id: Long): User {
        return userRepository.findById(id)
            .orElseThrow { IllegalArgumentException("User not found with id: $id") }
    }

    fun getUserByEmail(email: String): User? {
        return userRepository.findByEmail(email)
    }
}

data class CreateUserRequest(
    val email: String,
    val name: String
)